ts/parts/Axis.ts

```diff
@@ -5496,6 +5498,20 @@ extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */{
             }
             if (!tickPositionsOption && !tickPositioner) {
                 this.adjustTickAmount();
+
+                // Make sure ticks are within axis min and max range (#12716).
+                if (this.isOrdinal) {
+                    min = this.min;
+                    max = this.max;
+
+                    tickPositions = tickPositions.filter(
+                        (pos: number): boolean =>
+                            defined(min) && pos >= min &&
+                            defined(max) && pos <= max
```

Comment:
Instead of checking `defined(min)` in every iteration, can we don't call `filter` at all?